src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -120,15 +144,60 @@ void RaftReplDev::push_data_to_all_followers(repl_req_ptr_t rreq) {
 
     group_msg_service()
         ->data_service_request_unidirectional(nuraft_mesg::role_regex::ALL, PUSH_DATA, rreq->pkts)
-        .via(&folly::InlineExecutor::instance())
-        .thenValue([this, rreq = std::move(rreq)](auto e) {
+        .deferValue([this, rreq = std::move(rreq)](auto e) {
+            if (e.hasError()) {
+                RD_LOG(ERROR, "Data Channel: Error in pushing data to all followers: rreq=[{}] error={}",
+                       rreq->to_compact_string(), e.error());
+                handle_error(rreq, RaftReplService::to_repl_error(e.error()));
+                return;
+            }
             // Release the buffer which holds the packets
             RD_LOG(INFO, "Data Channel: Data push completed for rreq=[{}]", rreq->to_compact_string());
             rreq->fb_builder.Release();
             rreq->pkts.clear();
         });
 }
 
+void RaftReplDev::handle_error(repl_req_ptr_t const& rreq, ReplServiceError err) {
+    if (err == ReplServiceError::OK) { return; }
+
+    auto s = rreq->state.load();
+    if ((s & uint32_cast(repl_req_state_t::ERRORED)) ||
+        !(rreq->state.compare_exchange_strong(s, s | uint32_cast(repl_req_state_t::ERRORED)))) {
+        RD_LOG(INFO, "Raft Channel: Error in processing rreq=[{}] error={} already errored", rreq->to_compact_string(),
+               err);
+        return;
+    }
+
+    // Free the blks which is allocated already
```

Comment:
i believe in 'follower_create_req' function which calls 'do_blk_alloc', it also needs to check the status and call this handle_error, correct?

Reply:
Good point. Today that is not recoverable error and we crash. I am not sure we can handle the error. Perhaps we can do the discussion and handling as part of "Error recovery/validation" strategy.

Reply:
yes, sounds good